pkg/signup/signup_service.go

```diff
@@ -240,25 +242,42 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 }
 
 // GetUserSignup is used to return the actual UserSignup resource instance, rather than the Signup DTO
-func (s *ServiceImpl) GetUserSignup(userID string) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) GetUserSignup(userID string) (*v1alpha1.UserSignup, error) {
 	// Retrieve UserSignup resource from the host cluster
 	userSignup, err := s.UserSignups.Get(userID)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil, nil
-		}
 		return nil, err
 	}
 
 	return userSignup, nil
 }
 
 // UpdateUserSignup is used to update the provided UserSignup resource, and returning the updated resource
-func (s *ServiceImpl) UpdateUserSignup(userSignup *crtapi.UserSignup) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) UpdateUserSignup(userSignup *v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
 	userSignup, err := s.UserSignups.Update(userSignup)
 	if err != nil {
 		return nil, err
 	}
 
 	return userSignup, nil
 }
+
+func (s *ServiceImpl) CheckIfUserIsKnown(countryCode, phoneNumber string) error {
+	bannedUserList, err := s.BannedUsers.ListByPhoneNumber(countryCode + phoneNumber)
+	if err != nil {
+		return errors3.NewInternalError(err, "failed listing banned users")
+	}
+	if len(bannedUserList.Items) > 0 {
+		return errors3.NewForbiddenError("cannot re-register with phone number", "phone number already in use")
+	}
+
+	userSignupList, err := s.UserSignups.ListByPhoneNumber(countryCode + phoneNumber)
+	if err != nil {
+		return errors3.NewInternalError(err, "failed listing userSignups")
+	}
+	if len(userSignupList.Items) > 0 {
```

Comment:
Where do we check if the UserSignup found is not the user that is currently verifying?

Reply:
good point! fixed here: 707ae8b14f2c6d40e0c39c59f4ac656d4d19c320